src/tests/functional/plugin/shared/include/behavior/plugin/ov_auto_batching_tests.hpp

```diff
@@ -0,0 +1,165 @@
+        size_t,       // number of streams
+        size_t,       // number of requests
+        size_t>;      // batch size>
+
+class AutoBatching_Test : public OVPluginTestBase,
+                          public testing::WithParamInterface<AutoBatchTwoNetsParams> {
+public:
+    void SetUp() override {
+        std::tie(target_device, use_get_tensor, num_streams, num_requests, num_batch) = this->GetParam();
+        fn_ptrs = {ov::test::utils::make_single_conv(),
+                   ov::test::utils::make_multi_single_conv()};
+    };
+
+    static std::string getTestCaseName(const testing::TestParamInfo<AutoBatchTwoNetsParams> &obj) {
+        size_t streams, requests, batch;
+        bool use_get_tensor;
+        std::string target_device;
+        std::tie(target_device, use_get_tensor, streams, requests, batch) = obj.param;
+        return target_device + std::string(use_get_tensor ? "_get_tensor" : "_set_blob") + "_batch_size_" +
+               std::to_string(batch) +
+               "_num_streams_" + std::to_string(streams) + "_num_req_" + std::to_string(requests);
+    }
+
+protected:
+    bool use_get_tensor;
+    size_t num_streams;
+    size_t num_requests;
+    size_t num_batch;
+    std::vector<std::shared_ptr<ov::Model>> fn_ptrs;
+
+    void TestAutoBatch() {
+        auto core = ov::test::utils::PluginCache::get().core();
+
+        ov::OutputVector outputs;
+        std::vector<std::pair<std::shared_ptr<ov::Model>, ov::InferRequest>> irs;
+        std::vector<ov::InferRequest> irs_ref;
+        std::vector<int> outElementsCount;
+
+        for (size_t i = 0; i < fn_ptrs.size(); ++i) {
+            auto model = fn_ptrs[i];
+            auto inputs = model->inputs();
+            for (auto n : inputs) {
+                n.get_node()->set_output_type(0, ov::element::f32, n.get_shape());
+            }
+            ov::AnyMap config;
+            if (target_device.find("GPU") != std::string::npos) {
+                config[CONFIG_KEY(GPU_THROUGHPUT_STREAMS)] = std::to_string(num_streams);
+                config["INFERENCE_PRECISION_HINT"] = "f32";
+            }
+
+            if (target_device.find("CPU") != std::string::npos) {
+                config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = std::to_string(num_streams);
+                config[CONFIG_KEY(ENFORCE_BF16)] = CONFIG_VALUE(NO);
+            }
+            // minimize timeout to reduce test time
+            config[CONFIG_KEY(AUTO_BATCH_TIMEOUT)] = std::to_string(1);
+
+            auto compiled_model = core->compile_model(model, std::string(ov::test::utils::DEVICE_BATCH) + ":" +
+                                                      target_device + "(" + std::to_string(num_batch) + ")",
+                                                      config);
+
+            auto network_outputs = model->outputs();
+            ASSERT_EQ(network_outputs.size(), 1) << " Auto-Batching tests use networks with single output";
+            auto output = *network_outputs.begin();
```

Comment:
```suggestion
            const auto& output = network_outputs[0];
```
As assert above ensure that there is one output.

Reply:
fixed